### ISSUE TITLE
How can I attach to electron window?

### ISSUE THREAD

**Issue by DenisChulkov** [electron/spectron#255]
I test application which consists of standalone electron windows. Can I attach to electron window using spectron?

**Comment by santisantosh** [comment]
You can

this.app = new Application({
path=<<your exe path here>>
});

**Comment by DenisChulkov** [comment]
Good idea :)))) But in my case i dont have app path. Window already opened and i need attach to open window.

**Comment by santisantosh** [comment]
If you have more than 1 window, you can select any particular window. return this.app.client.getWindowCount()
Also
it('Focus on window',function () {
        this.app.client.windowHandles()
            .then((handles) => {
                const windowA = handles.values[0];
});
});

**Comment by DenisChulkov** [comment]
Hi @sahana2610. I found the way how to do it. Look into this link https://github.com/BorisOsipov/chromeless it's fork with "patch" which give a chance to attach to standalone electron windows.

**Comment by sahana2610** [comment]
@DenisChulkov : Thank you.  My electron app starts when I run the script in command line. I cannot make my tests run this script. Is there any way I can point out my test framework to this electron application without starting it through test?

**Comment by DenisChulkov** [comment]
@sahana2610 I handled electron window by title. You just need to investigate this fork and adapt it for your purposes.

**Comment by sahana2610** [comment]
@DenisChulkov Can you please give me a sample code of how you pointed it to the electron window by title.

**Comment by DenisChulkov** [comment]
Clone this fork. You can find sinple test as example inside.

чт, 7 июн. 2018 г., 16:29 sahana2610 <notifications@github.com>:

**Comment by sahana2610** [comment]
@DenisChulkov I am not able to find the link to the fork in your previous comment.

**Comment by sahana2610** [comment]
@DenisChulkov I went through the examples. I am still not able to figure out how you attached using the electron title. Did you skip the chromeless.goto and do like this: 
const chromeless = new Chromeless()
await chromeless.click('[title="Existing browser title"]')

**Comment by DenisChulkov** [comment]
@sahara2610 Also, you need debug port in application. If you dont have
debug port in your application you will not able to attach to the window.

вт, 26 июн. 2018 г. в 17:20, sahana2610 <notifications@github.com>:

**Comment by sahana2610** [comment]
@DenisChulkov Thank you. Will try doing that

**Comment by yetyman** [comment]
Hello, said code sample is no longer available, what did you do to enable attaching Spectron to existing electron windows?
